Open_Source/Section-3/3.1_Github_Best_Practices.md

```diff
@@ -0,0 +1,124 @@
+requests may land you in a big issue or conflict in the codebase. So, address and merge them at earliest. If the request is way too old to be addressed, tell the person who made the pull request to raise it again from a new branch. </br>
+
+git command to raise a pull request :
+```bash
+git push origin branch-name
+```
+This will push the changes into your forked repository.</br>
+After this head back to your GitHub repository and click on *Compare & pull request*, this will land you in a page where you need to give title and description to your PR and then tap *Create Pull Request*
+
+<div align="center">
+<img src="https://hisham.hm/img/posts/github-comparepr.png" width="600" >
+</div>
+</br>
+<div align="center">
+<img src="https://www.freecodecamp.org/news/content/images/2020/01/pullRequest-1.png" width="600" height="300" >
+</div>
+
+
+## 5. Stashing 
+Suppose while working on a branch you remembered that you need to work on another branch and you don't want to commit those changes in the first branch. No worries, *git stash* will help you sort that problem. It will save your work but not commit it, and you can come anytime on that branch and continue your work. </br>
+
+git stash command :</br>
+This command will add all the changes that are not commited, staged or unstaged to the stash pile.
+```bash
+git stash
+```
+This will reapply those changes that you stashed previously.
+```bash
+git stash pop
+```
+<div align="center">
+<img src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSPvJ6gLG2cMI8fNXVOo47oPh1tVNixBfxrTw&usqp=CAU" width="500" height="250">
+</div>
+
+
+## 6. Clean Commit Messages
+Write a short and meaningful commit messages. Add in the description the changes you made on the commit in detail. Doing so, will help in future if you needed to check out specific changes that were made, you will be able to distinguish the work and save a lot of time. 
+<div align="center">
+<img src="https://res.cloudinary.com/practicaldev/image/fetch/s--ZvkSCoQc--/c_imagga_scale,f_auto,fl_progressive,h_420,q_auto,w_1000/https://dev-to-uploads.s3.amazonaws.com/i/r1gk3rq3o70t3zghna8w.png" width="500" height="250">
+</div>
+
+## 7. Follow Community Guidlines
+The most imported thing is to follow the guidlines and the code of conduct provided by the organization you are contributing. Check for their naming convention on certain files.
+Pull request and commit message formats, guidlines are created so the worflow remains steady and no conflicts takes places while managing the contributed files/work. It can be a major hit if you don't follow their guidlines you may even get banned from contributing further to the project. So, always respect the organizations protocals.
+<div align="center">
+<img src="https://user-images.githubusercontent.com/77687018/133886437-ed0217de-c018-483c-98cc-d1e44ba8ac16.JPG" width="600" height="300">
+</div>
+
+<h5>
+  
+*Images reference -*

```

Comment:
Awesome & Creative @NamanManjkhola 🎉🎉
**Some changes you need to do:point_down:** 
Could you please remove those image refs, Since that does not look cool, and add `<hr>` at end of the doc.

Reply:
Yeah, sure